samples/notebooks/quickstart.ipynb

```diff
@@ -0,0 +1,572 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. Create a 'lightweight python component' from a Python function."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1.1 Define component function\n",
+    "The requirements for the component function:\n",
+    "* The function must be stand-alone.\n",
+    "* The function can only import packages that are available in the base image.\n",
+    "* If the function operates on numbers, the parameters must have type hints. Supported types are `int`, `float`, `bool`. Everything else is passed as `str`, that is, string.\n",
+    "* To build a component with multiple output values, use Python’s `typing.NamedTuple` type hint syntax."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def list_blobs(bucket_name: str) -> str:\n",
+    "  '''Lists all the blobs in the bucket.'''\n",
+    "  import subprocess\n",
+    "\n",
+    "  subprocess.call(['pip', 'install', '--upgrade', 'google-cloud-storage'])\n",
+    "  from google.cloud import storage\n",
+    "  storage_client = storage.Client()\n",
+    "  bucket = storage_client.get_bucket(bucket_name)\n",
+    "  list_blobs_response = bucket.list_blobs()\n",
+    "  blobs = ','.join([blob.name for blob in list_blobs_response])\n",
+    "  print(blobs)\n",
+    "  return blobs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1.2 Create a lightweight Python component"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import kfp.components as comp\n",
+    "\n",
+    "# Converts the function to a lightweight Python component.\n",
+    "list_blobs_op = comp.func_to_container_op(list_blobs)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1.3 Define pipeline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import kfp.dsl as dsl\n",
+    "\n",
+    "# Defines the pipeline.\n",
+    "@dsl.pipeline(name='List GCS blobs', description='Lists GCS blobs.')\n",
+    "def pipeline_func(bucket_name=dsl.PipelineParam('bucket')):\n",
+    "  list_blobs_task = list_blobs_op(bucket_name)\n",
+    "\n",
+    "# Compile the pipeline to a file.\n",
+    "import kfp.compiler as compiler\n",
+    "compiler.Compiler().compile(pipeline_func, 'list_blobs.pipeline.tar.gz')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2. Build a new Docker container image from a Python function.\n",
+    "Refer [here](https://runnable.com/docker/python/dockerize-your-python-application#alternatives) for more information."
```

Comment:
I'm not sure what this section is for. Can you explain a bit? This one doesn't seem to fit the theme of how to write components.

Reply:
Deleted.